### PR TITLE
fix merge_top and top operator bug for T_any Type

### PR DESCRIPTION
**Pull request by JackTan25**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6483

## What this PR does / why we need it:

### CODE REVIEW

**Review by fengttt — CHANGES_REQUESTED** [review]
This does not seems right.   When we will create a T_any type?   Actually should we EVER have a t_any type?

**Comment by JackTan25** [comment]
> This does not seems right. When we will create a T_any type? Actually should we EVER have a t_any type?

Yes, please see  #6483 and my tests. These give the examples.

**Comment by JackTan25** [comment]
@matrix-meow rerun all failed

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

File: pkg/sql/colexec/top/types.go [hunk 1]
```diff
@@ -60,6 +64,11 @@ func (ctr *container) cleanBatch(mp *mpool.MPool) {
 
 func (ctr *container) compare(vi, vj int, i, j int64) int {
 	for _, pos := range ctr.poses {
+		// for order by cols, if there is a T_any type in cols,
+		// we won't init a compare
+		if ctr.cmps[pos] == nil {
```
Comment (nnsgmsone) [inline pkg/sql/colexec/top/types.go:69]: If set to nil. Does it not participate in the sorting?

**Pull request closed without merging**